src/app/content/highlights/reducer.ts

```diff
@@ -41,6 +43,11 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
     case getType(locationChange): {
       // Noops for locationChange dispatched when search query changes
       if (action.payload.action === 'REPLACE') { return state; }
+
+      const summaryShouldBeOpen =
+        getParamFromQuery(modalQueryParameterName, action.payload.location.search) === modalUrlName
+        && action.payload.action === 'PUSH';
```

Comment:
if modals are causing a location change do you even need this logic to close it? this reducer already closes the dialog on any location change. with the `push` condition in here i'm not sure its even doing anything

Reply:
This is for preventing them from closing on location change since it hits the reducer after `open<Modal>`

Reply:
i don't understand, are you saying that there is something that is undoing a previous acton if you don't have this check here?

Reply:
I think he meant that `openMyHighlights` is opening a modal, and then there is a `openModal` hook attached to this action that calls dispatches `push` action and then this `push` action is reduced in here as `locationChange` so we check if we should or should not close the modal

Reply:
why would we need the push filter? in that case the location search would include this modal name and this reducer would set `summaryShouldBeOpen` to `true`, that can be the thing that actually opens it, we don't need to reduce `openMyHighlights` (this brings us to supporting links and the forward button that open the dialogs)

Reply:
I'll remove reducing `openMyHighlights` and `closeMyHighlights` etc.
We filter here for `PUSH` because we wanted to handle opening a page with `modal` param (and support for forward button) on a separate card (it requires us to do more changes like: initialize load summary, handle modal=SG for book that does not have them, send GA events about opening modals, etc.)

Reply:
Actually removing reducer for `openMyHighlights` / `PQ`, `SG` breaks like 20+ tests. These are tests like: open mh, check if modal is rendered - we would have to replace the calls with locationChange dispatch, but there are also tests like "click print highlights > start loading all of them > close modal while they are loading > make sure that we do not print them"
If we remove the reducer for `closeMyHighlights` then we would have to call `locationChange` without modal param instead of calling a close modal action because unit test will not know how to handle `.goBack` and also `createTestStore` does not implement middlewares do hooks are not supported by default.

I'm not sure if all of these changes are in the scope of this issue

Reply:
We can't because we are now opening a modal using `dispatch(openMyHighlights)` which is setting `open: true` and then there is a `openModalHook` that dispatches `push`/`locationChange` to update the history state and url. 

I think the proper solution would be to remove reducers for `openMyHighlights` and `closeMyHighlights` actions and handle opening modal using only `locationChange` action as you suggested, but this generates a tons of errors and I think we should do this in another card

Reply:
ah so the purpose of this logic is really just to ignore the location change triggered by the open modal action where it adds the query param to the url. we would prefer to have that location change open the dialog, but we're deferring that to another card.

which card will that be done in?

Reply:
https://app.zenhub.com/workspaces/openstax-unified-5b71aabe3815ff014b102258/issues/openstax/unified/1389

Reply:
i added a note to that card about removing the action type filtering